misk/src/main/kotlin/misk/web/jetty/ConnectionMetricsCollector.kt

```diff
@@ -0,0 +1,37 @@
+package misk.web.jetty
+
+import org.eclipse.jetty.io.Connection
+import org.eclipse.jetty.util.component.AbstractLifeCycle
+
+internal class ConnectionMetricsCollector(
+  protocol: String,
+  port: Int,
+  private val metrics: ConnectionMetrics
+) : AbstractLifeCycle(), Connection.Listener {
+
+  private val labels = ConnectionMetrics.forPort(protocol, port)
+
+  override fun onOpened(connection: Connection) {
+    metrics.activeConnections.labels(*labels).inc()
+    metrics.acceptedConnections.labels(*labels).inc()
+  }
+
+  override fun onClosed(connection: Connection) {
+    metrics.activeConnections.labels(*labels).dec()
+
+    val bytesIn = connection.bytesIn
+    if (bytesIn > 0) metrics.bytesReceived.labels(*labels).inc(bytesIn.toDouble())
+
+    val bytesOut = connection.bytesOut
+    if (bytesOut > 0) metrics.bytesSent.labels(*labels).inc(bytesOut.toDouble())
+
+    val messagesIn = connection.messagesIn
+    if (messagesIn > 0) metrics.messagesReceived.labels(*labels).inc(messagesIn.toDouble())
+
+    val messagesOut = connection.messagesOut
+    if (messagesOut > 0) metrics.messagesSent.labels(*labels).inc(messagesOut.toDouble())
+
+    val connectionDuration = (System.currentTimeMillis() - connection.createdTimeStamp).toDouble()
```

Comment:
createdTimeStamp ugh

Reply:
interesting that we don’t count connections ’til they’re closed

Reply:
(no idea on how to do better here, but it could be surprising)

Reply:
Yeah it's weird (although also what we have today). We could potentially do something like track the connections and have a periodic refresh timer that updates the counters based on this. Let me add that.